src/react-integration/__tests__/hooks.tsx

```diff
@@ -286,6 +290,27 @@ describe('useRetrieve', () => {
     rerender();
     expect(dispatch).toBeCalled();
   });
+  it('should dispatch with resource defined dataExpiryLength', async () => {
+    function FetchTester() {
+      useRetrieve(StaticArticleResource.singleRequest(), payload);
```

Comment:
this only tests the dispatch side; I would really appreciate some tests that touch the network manager side - hopefully that actually test the caching length.

At the very least, the `integration.tsx` test file should get a full path done so the code coverage doesn't decrease.

Reply:
Added test cases for the middleware from NetworkManager. Covers fetch actions success/error as well as the expiry time with and without overridden dataExpiryLength or errorExpiryLength.